perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/impl/AppAutoRejectionScheduler.java

```diff
@@ -0,0 +1,252 @@
+package cz.metacentrum.perun.registrar.impl;
+
+import cz.metacentrum.perun.core.api.Attribute;
+import cz.metacentrum.perun.core.api.BeansUtils;
+import cz.metacentrum.perun.core.api.ExtSourcesManager;
+import cz.metacentrum.perun.core.api.Group;
+import cz.metacentrum.perun.core.api.PerunClient;
+import cz.metacentrum.perun.core.api.PerunPrincipal;
+import cz.metacentrum.perun.core.api.PerunSession;
+import cz.metacentrum.perun.core.api.Vo;
+import cz.metacentrum.perun.core.api.exceptions.GroupNotExistsException;
+import cz.metacentrum.perun.core.api.exceptions.InternalErrorException;
+import cz.metacentrum.perun.core.api.exceptions.PerunException;
+import cz.metacentrum.perun.core.api.exceptions.VoNotExistsException;
+import cz.metacentrum.perun.core.bl.PerunBl;
+import cz.metacentrum.perun.core.impl.Synchronizer;
+import cz.metacentrum.perun.registrar.RegistrarManager;
+import cz.metacentrum.perun.registrar.model.Application;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.jdbc.core.JdbcPerunTemplate;
+
+import javax.sql.DataSource;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Class handling auto rejection of expired applications for VOs and Groups
+ *
+ * @author Jakub Hejda <Jakub.Hejda@cesnet.cz>
+ */
+public class AppAutoRejectionScheduler {
+
+	private final static Logger log = LoggerFactory.getLogger(Synchronizer.class);
+
+	private JdbcPerunTemplate jdbc;
+	private PerunSession sess;
+	private PerunBl perun;
+	private RegistrarManager registrarManager;
+
+	private static final String A_VO_APP_EXP_RULES = "urn:perun:vo:attribute-def:def:applicationExpirationRules";
+	private static final String A_GROUP_APP_EXP_RULES = "urn:perun:group:attribute-def:def:applicationExpirationRules";
+
+	@Autowired
+	public void setDataSource(DataSource dataSource) {
+		this.jdbc = new JdbcPerunTemplate(dataSource);
+		this.jdbc.setQueryTimeout(BeansUtils.getCoreConfig().getQueryTimeout());
+	}
+
+	public PerunBl getPerun() {
+		return perun;
+	}
+
+	@Autowired
+	public void setPerun(PerunBl perun) {
+		this.perun = perun;
+	}
+
+	public RegistrarManager getRegistrarManager() { return registrarManager; }
+
+	@Autowired
+	public void setRegistrarManager(RegistrarManager registrarManager) { this.registrarManager = registrarManager; }
+
+	/**
+	 * Constructor for unit tests
+	 *
+	 * @param perun PerunBl bean
+	 */
+	public AppAutoRejectionScheduler(PerunBl perun) {
+		this.perun = perun;
+		initialize();
+	}
+
+	public void initialize() {
+		String synchronizerPrincipal = "perunSynchronizer";
```

Comment:
I think this is not perunSynchronizer :)

Reply:
Changed to perunRegistrar.